File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer,Integer> m=new HashMap<>();
        int x[]=new int[2];
        for(int i=0;i<nums.length;i++){
            m.put(nums[i],i);
        }
        for(int i=0;i<nums.length;i++){
            if( m.containsKey(target-nums[i]) && m.get(target-nums[i])!=i ){
                x[0]=i;
                x[1]=m.get(target-nums[i]);
                break;
            }
        }
        return x;
    }
}

File: 1-two-sum/two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        d={}
        for i in range(0,len(nums)):
            d[nums[i]]=i
        for i in range(0,len(nums)):
            if target-nums[i] in d.keys() and i!=d[target-nums[i]]:
                return [i,d[target-nums[i]]]
        return[-1,-1]
        